The target detection matching the vendor "unknown" gets
arm musl targets wrong. Add a match against the raw_os,
which is linux-musleabihf to get a better match.

`assert_rust_compile` directly calls `rustc` without applying RUSTFLAGS.
Without taht, the --sysroot flag is missing when cross-compiling.
Just skip the check, we know the compiler works anyways.
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -372,6 +373,9 @@
                 ]
                 if not matches:
                     continue
+                for c in matches:
+                    if c.target.raw_os == host_or_target.raw_os:
+                        return c.rust_target
                 # As below, correlate on the (normalized) vendor.
                 for c in matches:
                     if c.target.vendor == vendor:
@@ -529,7 +529,6 @@
 @checking("for rust target triplet")
 def rust_target_triple(rustc, target, rust_supported_targets, arm_target):
     rustc_target = detect_rustc_target(target, arm_target, rust_supported_targets)
-    assert_rust_compile(target, rustc_target, rustc)
     return rustc_target
 
 
